lib/IRGen/IRGenSIL.cpp

```diff
@@ -2472,6 +2476,76 @@ static void emitDynamicSelfMetadata(IRGenSILFunction &IGF) {
   IGF.setDynamicSelfMetadata(selfTy, isExact, value, selfKind);
 }
 
+namespace {
+// Notes the use of foreign types in generic arguments for C++ interop.
+class NoteUseOfForeignTypeMetadata
```

Comment:
Why not use Type::walk() to recursively visit everything instead of manually destructuring types like this?

Reply:
Oh, I totally missed the existence of that API! Thanks for bringing it to my attention. I'll give this a try tomorrow.

Reply:
Thanks! I switched to `Type::walk` and it looks much cleaner.